analyze_nplus_R30_directives.go

```diff
@@ -0,0 +1,2112 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// All the definitions are extracted from the source code
+// Each bit mask describes these behaviors:
+//   - how many arguments the directive can take
+//   - whether or not it is a block directive
+//   - whether this is a flag (takes one argument that's either "on" or "off")
+//   - which contexts it's allowed to be in
+
+package crossplane
+
+//nolint:gochecknoglobals
+var ngxPlusR30Directives = map[string][]uint{
```

Comment:
from the code, 
NgxPlusR31DirectivesMatchFn has the list of all directives supported until R31 and not purely only those introduced with R31? So,

To validate my config against a cutoff version - support everything <=R30
DirectiveSources:         []MatchFunc{NgxPlusR30DirectivesMatchFn}

To validate my config against a cutoff version - support everything <=R31
DirectiveSources:         []MatchFunc{NgxPlusR31DirectivesMatchFn}

If there are overlaps, we would pass the directive based on its first match, we ignore the order?
DirectiveSources:         []MatchFunc{NgxPlusR30DirectivesMatchFn, NgxPlusR31DirectivesMatchFn}

Reply:
> NgxPlusR31DirectivesMatchFn has the list of all directives supported until R31 and not purely only those introduced with R31?

Nah, it only has directives in R31 source code, not all directives supported until R31. If a directive appears in R31 source code it should appear in `NgxPlusR31Directives`, otherwise not.

> To validate my config against a cutoff version - support everything <=R30

For this case we should go in this way: DirectiveSources: []MatchFunc{NgxPlusR30DirectivesMatchFn, NgxPlusR29DirectivesMatchFn, NgxPlusR28DirectivesMatchFn,...}. I assumed we only need the latest 3 versions, but if needed I can add MatchFn for R29, R28, or more.

> If there are overlaps, we would pass the directive based on its first match, we ignore the order?
DirectiveSources: []MatchFunc{NgxPlusR30DirectivesMatchFn, NgxPlusR31DirectivesMatchFn}

We ignore the order, get all bitmasks from all DirectiveSources. If a directive aligns with any bitmask from any of them, it can pass the validation. For example, say `absolute_redirect` has bitmask1,bitmask2 in NgxPlusR30DirectivesMatchFn, bitmask3 in NgxPlusR31DirectivesMatchFn. It can pass the validation if it satisfy bitmask1, bitmask2, or bitmask3

Reply:
`analyze_nplus_R30_directives.go` has all directives up until R30 and hence I inferred
To validate my config against a cutoff version - support everything <=R30
`DirectiveSources: []MatchFunc{NgxPlusR30DirectivesMatchFn}`